### PR TITLE
Loading local Dapps from FS.

### LINKED ISSUE #1183: Dapps from local filesystem
make `http://fs.parity/home/gav/Chicken/index.html` work like magic.

also consider making `http://local.parity/Chicken/index.html` act equivalently to `http://fs.parity/home/gav/.parity/dapps/Chicken/index.html`.

### PR DESCRIPTION
**Pull request by tomusdrw**
Closes #1183

### CODE REVIEW

File: dapps/src/page/builtin.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		let info = app.info();
+		PageEndpoint {
+			app: Arc::new(app),
+			prefix: None,
+			safe_to_embed: false,
+			info: EndpointInfo::from(info),
+		}
+	}
+
+	pub fn with_prefix(app: T, prefix: String) -> Self {
```
Comment (gavofyork) [inline dapps/src/page/builtin.rs]: undocumented `pub`?

File: dapps/src/page/builtin.rs [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	pub app: Arc<T>,
+	/// Prefix to strip from the path (when `None` deducted from `app_id`)
+	pub prefix: Option<String>,
+	/// Safe to be loaded in frame by other origin. (use wisely!)
+	safe_to_embed: bool,
+	info: EndpointInfo,
+}
+
+impl<T: WebApp + 'static> PageEndpoint<T> {
+	pub fn new(app: T) -> Self {
```
Comment (gavofyork) [inline dapps/src/page/builtin.rs]: undocumented `pub`?

Comment (tomusdrw) [inline dapps/src/page/builtin.rs]: It's only module-pub, so no warnings. But will write some docs.

**Comment by gavofyork** [comment]
([ci skip] could have been invoked as the changes added were only comments.)

**Pull request merged**